Instruments/microwave_tuning_gui.py

```diff
@@ -134,18 +135,17 @@ def generate_data(self):
                 self.slider_min.value(),
                 "slider max",
                 self.slider_max.value())
-        if len(self.line_data) > 6:
-            self.line_data.pop(0)
-            self.x.pop(0)
-        self.x.append(np.r_[
-                self.slider_min.value():
-                self.slider_max.value():
-                15j])
+        a = self.slider_min.value()
+        b = self.slider_max.value()
+        if hasattr(self.B12,'freq_bounds'):
+            a = a if a > np.ceil(self.B12.freq_bounds[0]/1e3) else np.ceil(self.B12.freq_bounds[0]/1e3)
+            b = b if b < np.floor(self.B12.freq_bounds[-1]/1e3) else np.floor(self.B12.freq_bounds[-1]/1e3)
+        self.x.append(np.r_[a:b:15j])
```

Comment:
This is setting the new frequency sweep values after the sliders have been adjusted

Reply:
This was in place so that the MWtune is used to capture the dip at a set width and no wider, you should either be recapturing with current settings or zooming in (no reason to go below 9.815 GHz or above 9.825 GHz with only 15 steps) so the dip can be seen with 15 points

Reply:
explain in code with succinct comment

Reply:
we decided there was some confusion over frq sweep and lock, and that this wasn't wanted